flytestdlib/storage/storage.go

```diff
@@ -98,6 +98,10 @@ type ReferenceConstructor interface {
 	// ConstructReference creates a new dataReference that matches the storage structure.
 	ConstructReference(ctx context.Context, reference DataReference, nestedKeys ...string) (DataReference, error)
 
+	// Explain that in order for this interface to be part of a CRD we need to implement a DeepCopy method as per
+	// https://pkg.go.dev/k8s.io/gengo/examples/deepcopy-gen.
+	DeepCopyReferenceConstructor() ReferenceConstructor
```

Comment:
@EngHabu , as we spoke offline, we have to add this method as part of the interface.